client/app/controllers/shows.controller.js

```diff
@@ -4,11 +4,27 @@ angular.module('Tour-Track')
 .controller('ShowsCtrl', ['$scope', 'allShows', 'ShowFactory', 'VenueFactory', '$timeout', function($scope, allShows, ShowFactory, VenueFactory, $timeout) {
 
   var years = [];
+  $scope.filter = { date: "", venue: "", location: "" };
 
-  var Shows = function() {
+  $scope.$watch('filter', (filter) => {
+    years = [];
+    var dateString = filter.date ? filter.date.toISOString().slice(0,10) : ""
+
+    var filteredShows = allShows.filter( (show) => {
+      return show.date.slice(0,10).indexOf(dateString) > -1 
+        && show.location.toLowerCase().indexOf(filter.location.toLowerCase()) > -1 
+        && show.venue_name.toLowerCase().indexOf(filter.venue.toLowerCase()) > -1;
```

Comment:
Excessive nested loops, all three of these could be condensed into one. 1/3 the iterations.